hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -2028,7 +2028,8 @@ private ReplicaInfo finalizeReplica(String bpid, ReplicaInfo replicaInfo)
       } else {
         FsVolumeImpl v = (FsVolumeImpl)replicaInfo.getVolume();
         if (v == null) {
-          throw new IOException("No volume for block " + replicaInfo);
+          throw new IOException("No volume for block " + replicaInfo +
```

Comment:
`throw new IOException("No volume for bpid: " + bpid + " , block: " + replicaInfo););`